pkg/services/eventstore/otel/factory.go

```diff
@@ -0,0 +1,245 @@
+// pkg/services/eventstore/otel/factory.go
+package otel
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/qpoint-io/qtap/pkg/buildinfo"
+	"github.com/qpoint-io/qtap/pkg/config"
+	"github.com/qpoint-io/qtap/pkg/services"
+	"github.com/qpoint-io/qtap/pkg/services/eventstore"
+	"github.com/qpoint-io/qtap/pkg/services/objectstore"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutlog"
+	sdklog "go.opentelemetry.io/otel/sdk/log"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+	"go.uber.org/zap"
+	"golang.org/x/term"
+)
+
+// Sample config:
+// eventstore:
+//   - type: otel
+//     endpoint: "localhost:4317"  # OTLP endpoint (gRPC: 4317, HTTP: 4318)
+//     protocol: grpc              # "grpc", "http", or "stdout"
```

Comment:
```suggestion
//     exporter: otlp              # "otlp", "console"
//     protocol: grpc              # "grpc", "http/protobuf"
```

we should match the [otel exporter spec](https://opentelemetry.io/docs/specs/otel/protocol/exporter/#specify-protocol)

example from https://pkg.go.dev/go.opentelemetry.io/contrib/exporters/autoexport#NewLogExporter
```go
// OTEL_LOGS_EXPORTER defines the logs exporter; supported values:
//   - "none" - "no operation" exporter
//   - "otlp" (default) - OTLP exporter; see [go.opentelemetry.io/otel/exporters/otlp/otlplog]
//   - "console" - Standard output exporter; see [go.opentelemetry.io/otel/exporters/stdout/stdoutlog]
//
// OTEL_EXPORTER_OTLP_PROTOCOL defines OTLP exporter's transport protocol;
// supported values:
//   - "http/protobuf" (default) -  protobuf-encoded data over HTTP connection;
//     see: [go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp]
//   - "grpc" - gRPC with protobuf-encoded data over HTTP/2 connection;
//     see: [go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc]
```

Reply:
this confuses me. I don't understand why we would have two properties when one will do clearly.

Reply:
No worries - I thought it would be clearer to match the common otel exporter config format but it's not that important in here.